src/components/vm/disks/diskAdd.jsx

```diff
@@ -513,11 +514,49 @@ export class AddDiskModalBody extends React.Component {
             break;
         }
         case 'file': {
+            this.setState({ file: value });
+
             if (value.endsWith(".iso")) {
                 // use onValueChange instead of setState in order to perform subsequent state change logic
                 this.onValueChanged("device", "cdrom");
             }
-            this.setState({ file: value });
+
+            if (value) {
+                cockpit.spawn(["/bin/bash", "-c", "xargs -0n1 head --bytes=16 "], { binary: true, err: "message", superuser: "try" })
+                        .input(value)
+                        .then(output => {
+                            console.log(output);
+                            let format = "";
+                            const file_identifier = Array.from(output).slice(0, 4)
+                                    .map(function(i) { return '0x' + i.toString(16) })
```

Comment:
Use a proper arrow here, i.e. `.map(i= > '0x' ...)`.

Moreover, do we really need to do this conversion? Can't you just directly compare the first 4 entries of `output` to `[81, 70, 73, 251]`? (one by one of course, as list comparisons don't work in JS). That would avoid a lot of back and forth, and it's also not clear how other future formats look like -- they may not have a similar magic in the first 4 bytes.